src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_actions.py

```diff
@@ -163,7 +163,8 @@ def _handle_auth_types(**kwargs):
         if args.ssh_dest_key_path:
             raise CLIError('SSH parameters cannot be used with password authentication type')
         elif not args.admin_password:
-            raise CLIError('Admin password is required with password authentication type')
+            import getpass
+            args.admin_password = getpass.getpass('Admin Password: ')
```

Comment:
i will open an issue that we should still error out when running under scripting mode.